src/main/java/core/basesyntax/strategy/OperationHandler.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public interface OperationHandler {
+    int MIN_QUANTITY = 0;
+    void applyOperation(FruitTransaction transaction);
+
+    default Integer getCurrentBalance(FruitTransaction transaction) {
+        return Storage.reports.get(transaction.getFruitName());
```

Comment:
Direct access to the `Storage.reports` breaks encapsulation. Consider providing a method in the `Storage` class to encapsulate the access to the reports map.